### PR TITLE
Respect deps.json part 2

### LINKED ISSUE #4081: Consider adopting ComponentDependencyResolver for tasks on Core
See https://github.com/dotnet/corefx/issues/33165. This is designed for plugin-type contexts, just like we have.

### PR DESCRIPTION
**Pull request by Forgind**
Fixes #4081

### Context
See https://github.com/dotnet/msbuild/pull/7520

### Changes Made
Added looking for a .deps.json

### Testing
Found one case when it seemed to discard an assembly without a .deps.json and one when it seemed to include it. (Not extensive)

### CODE REVIEW

File: src/Shared/MSBuildLoadContext.cs [hunk 1]
```diff
@@ -31,6 +36,10 @@ public MSBuildLoadContext(string assemblyPath)
             : base($"MSBuild plugin {assemblyPath}")
         {
             _directory = Directory.GetParent(assemblyPath)!.FullName;
+
+            _resolver = File.Exists(assemblyPath) && File.Exists(Path.Combine(_directory, Path.GetFileNameWithoutExtension(assemblyPath) + ".deps.json"))
```
Comment (rainersigwald) [inline src/Shared/MSBuildLoadContext.cs:40]:
```suggestion
            _resolver = File.Exists(assemblyPath) && File.Exists(Path.ChangeExtension(assemblyPath) + ".deps.json"))
```

?

Comment (rainersigwald) [inline src/Shared/MSBuildLoadContext.cs:40]: Comment both parts of this condition, please?

**Review by rainersigwald — COMMENTED** [review]
Have you validated this code against the NuGet workflow that failed?

cc @baronfel

**Review by rainersigwald — APPROVED** [review]

**Comment by Forgind** [comment]
> Oh and let's make sure to squash this before merge; the commit history is super confusing with the revert.

Do you mean make sure to use squash and merge when merging, or that I should rebase this into a few interesting commits?

**Pull request merged**